### PR TITLE
fix php 7.2 error : Countable interface not implemented

### PR DESCRIPTION
**Pull request by gmachure**
### Description (*)

We created a custom module with ui component form with a file upload field. While the upload, the method _setUploadFileId from vendor/magento/framework/File/Uploader.php generate a warning because Countable interface is not implement for a string (PHP version : 7.2.12).
$file[0] is a string and not an array. So count($file[0]) generate an error.

### Fixed Issues (if relevant)

1. #19247: fix php 7.2 error : Countable interface not implemented

### Manual testing scenarios (*)

1. Upload image in ui component form in a custom module
2. Test :
-Create custom module
-Add a file uploader in ui component form.
-Upload an image.
-PHP version : 7.2.12

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @gmachure. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by gmachure** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @gmachure. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @gmachure, here is your [new Magento instance](http://34.228.235.121/pr-19249/).
Admin access: [http://34.228.235.121/pr-19249//admin](http://34.228.235.121/pr-19249//admin)
Login: `admin` Password: `123123q`

File: lib/internal/Magento/Framework/File/Uploader.php [hunk 1]
```diff
@@ -546,7 +546,7 @@ private function _setUploadFileId($fileId)
 
             preg_match("/^(.*?)\[(.*?)\]$/", $fileId, $file);
 
-            if (is_array($file) && count($file) > 0 && count($file[0]) > 0 && count($file[1]) > 0) {
+            if (is_array($file) && count($file) > 0 && !empty($file[0]) && (isset($file[1]) && !empty($file[1]))) {
```
Comment (orlangur) [inline lib/internal/Magento/Framework/File/Uploader.php]: `isset($file[1])` is redundant here, please remove it.

**Review by orlangur — CHANGES_REQUESTED** [review]

File: lib/internal/Magento/Framework/File/Uploader.php (hunk 1 above)
Comment (gmachure) [reply lib/internal/Magento/Framework/File/Uploader.php]: Hi @orlangur 
I removed redundant `isset($file[1])`

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-3483](https://jira.corp.magento.com/browse/ENGCOM-3483) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@gmachure thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by magento-engcom-team** [comment]
Hi @gmachure. Thank you for your contribution. 
We will aim to release these changes as part of 2.3.1. 
Please check the release notes for final confirmation.

**Pull request merged**